### DISCUSSION TITLE
New versioning & release approach

### DISCUSSION THREAD

**Discussion by bdbch** [ueberdosis/tiptap#3197] [Announcements]
Hey Tiptap community!

After an unfortunate longer break we're able to work on Tiptap again. The first thing we wanted to solve is the convoluted versioning for our packages. In the past, it was really hard to figure out which version of an Tiptap extension is compatible with the current core version installed. To combat this, we started to release packages non-individually which means all versions of our packages are getting bumped to the most recent version.

This means versions went from

```
@tiptap/package-a@2.0.0-beta.89
@tiptap/package-b@2.0.0-beta.184
@tiptap/package-c@2.0.0-beta.32
```

to

```
@tiptap/package-a@2.0.0-beta.184
@tiptap/package-b@2.0.0-beta.184
@tiptap/package-c@2.0.0-beta.184
```

This will mean we will have version bumps for packages that don't introduce new functionalities, fixes etc.

We also plan to do more frequent releases to introduce fixes and new features faster to the editor. For the keen eyes you maybe also realized we're having a single CHANGELOG.md in our projects root now and we're back to using Github Releases.

This means you will be able to see all changes of a release in one place.

You can read the example changelog for our latest release **2.0.0-beta.195** [here](https://github.com/ueberdosis/tiptap/releases/tag/v2.0.0-beta.195).

Let us know your thoughts in this discussion.

**Comment by rfgamaral** [comment]
This is awesome to see, great work @bdbch (and Tiptap team overall). Kudos 👏

**Reply by bdbch** [reply]
Thank you @rfgamaral! Let's see how this change will work out in the next few days. 🥳

As soon as we finally hit a stable release version, we'll also introduce prerelease tags. Right now we're releasing those beta versions as "non-prereleases" as they won't show up on the Github sidebar by default and I think having more exposure on the changelog helps a lot.

**Comment by Korywon** [comment]
Are we considering on versioning the pro extensions the same way as well? The emoji version is still on `2.0.0-beta.17`.

**Reply by bdbch** [reply]
Yes we're working on that too! In general we need to do a lot of changes we did for the non-pro packages for the pro packages.

**Reply by Korywon** [reply]
Sounds like a plan! Do you guys happen to have a rough timeline for the pro packages? Asking since I'm wondering if it might solve #3206. 🤔

**Reply by svenadlung** [reply]
@Korywon It could IMO. If I am not wrong, the changes are also related to the typings. We will figure out soon.

**Comment by benatshippabo** [comment]
Any timeline on when the v2 stable release will be cut?

**Reply by nperez0111** [reply]
Released!

**Reply by benatshippabo** [reply]
You're the best @nperez0111 ty
